### PR TITLE
hook up apptool events

### PR DESCRIPTION
**Pull request by denis-yuen**
**Description**
Hook up app tool events to events so they can show up in the recent activities dashboard
Requires a corresponding UI PR

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-4041

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] Check the Snyk dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java [hunk 1]
```diff
@@ -261,13 +263,7 @@ public Collection addEntryToCollection(@ApiParam(hidden = true) @Parameter(hidde
...
                 .withType(Event.EventType.ADD_TO_COLLECTION);
 
-        if (entryAndCollection.getLeft() instanceof BioWorkflow) {
-            eventBuild = eventBuild.withBioWorkflow((BioWorkflow)entryAndCollection.getLeft());
-        } else if (entryAndCollection.getLeft() instanceof Service) {
-            eventBuild = eventBuild.withService((Service)entryAndCollection.getLeft());
-        } else if (entryAndCollection.getLeft() instanceof Tool) {
-            eventBuild = eventBuild.withTool((Tool)entryAndCollection.getLeft());
-        }
+        eventBuild = modifyBuilderForCollection(entryAndCollection, eventBuild);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java:266]: I think you made this same comment on an earlier PR of mine, but you can instead have on lines 258/260 and 316/312

```
Event.Builder eventBuild = entryAndCollection.getLeft().getEventBuilder()
  .withOrganization(...
```

**Review by coverbeck — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java (hunk 1 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java]: past me had good tips

File: dockstore-webservice/src/main/resources/migrations.1.12.0.xml [hunk 2]
```diff
@@ -387,4 +387,10 @@
             alter table sourcefile enable row level security;
         </sql>
     </changeSet>
+    <changeSet author="dyuen (generated)" id="addAppToolEvents">
+        <addColumn tableName="event">
+            <column name="apptoolid" type="int8"/>
```
Comment (svonworl) [inline dockstore-webservice/src/main/resources/migrations.1.12.0.xml:392]: Out of curiosity, is there ever a practical difference between `int8` and `bigint`?  (I don't know the answer.)

**Comment by svonworl** [comment]
Random feeling: I wish the creators of `ImmutablePair` had picked different names for the constituent values, like 'A' and 'B' or 'First' and 'Second'.  For whatever reason, `getLeft` and `getRight` make me dizzy.

**Comment by svonworl** [comment]
Should we remove the many-argumented `Event` constructor, since we seem to use the `Builder` everywhere?

**Review by svonworl — APPROVED** [review]

**Comment by denis-yuen** [comment]
> Should we remove the many-argumented `Event` constructor, since we seem to use the `Builder` everywhere?

probably

File: dockstore-webservice/src/main/resources/migrations.1.12.0.xml (hunk 2 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/resources/migrations.1.12.0.xml:392]: Nope, looks like its just an alias https://www.postgresql.org/docs/12/datatype-numeric.html

**Review by NatalieEO — APPROVED** [review]

**Comment by denis-yuen** [comment]
Followed by https://ucsc-cgl.atlassian.net/browse/SEAB-4055

**Pull request merged**